core/field_dropdown.ts

```diff
@@ -414,6 +399,30 @@ export class FieldDropdown extends Field<string> {
     return this.generatedOptions;
   }
 
+  /**
+   * Update the options on this dropdown. This will reset the selected item to
+   * the first item in the list.
+   *
+   * @param menuGenerator The array of options or a generator function.
+   */
+  setOptions(menuGenerator: MenuGenerator) {
+    if (Array.isArray(menuGenerator)) {
+      this.validateOptions(menuGenerator);
+      const trimmed = this.trimOptions(menuGenerator);
+      this.menuGenerator_ = trimmed.options;
+      this.prefixField = trimmed.prefix || null;
+      this.suffixField = trimmed.suffix || null;
+    } else {
+      this.menuGenerator_ = menuGenerator;
+    }
+    /**
```

Comment:
Nit: this shouldn't be a doc comment. I know you copied it over, but might as well fix it while here.